src/main/java/com/cleanroommc/groovyscript/helper/ingredient/IngredientHelper.java

```diff
@@ -86,6 +87,18 @@ public static boolean isEmpty(@Nullable NBTTagCompound nbt) {
         return nbt == null || nbt.isEmpty();
     }
 
+    public static boolean overMaxSize(@Nullable IIngredient ingredient, int maxSize) {
```

Comment:
what about adding methods to directly add to the `msg` and calling those?
additionally, `IResourceStack` should probably be used here instead of `IIngredient`, since `IResourceStack` is the interface that has `getAmount`.
```java
public static void overMaxSize(GroovyLog.Msg msg, Collection<IIngredient> ingredients, int maxSize) {
    if (!GroovyScriptConfig.compat.checkInputStackCounts) return;
    for (var ingredient : ingredients) {
        msg.add(IngredientHelper.overMaxSize(ingredient, maxSize), "Expected stack size of {} for {}, got {}", maxSize, ingredient, ingredient.getAmount());
    }
}

public static void overMaxSize(GroovyLog.Msg msg, IIngredient ingredient, int maxSize) {
    if (!GroovyScriptConfig.compat.checkInputStackCounts) return;
    msg.add(IngredientHelper.overMaxSize(ingredient, maxSize), "Expected stack size of {} for {}, got {}", maxSize, ingredient, ingredient.getAmount());
}
```